addons/web/static/tests/_framework/env_test_helpers.js

```diff
@@ -0,0 +1,114 @@
+/** @odoo-module */
+
+import { after, registerDebugInfo } from "@odoo/hoot";
+import { on } from "@odoo/hoot-dom";
+import { createDebugContext } from "@web/core/debug/debug_context";
+import { registry } from "@web/core/registry";
+import { makeEnv, startServices } from "@web/env";
+import { MockServer, makeMockServer } from "./mock_server/mock_server";
+
+/**
+ * @typedef {import("@web/env").OdooEnv} OdooEnv
+ *
+ * @typedef {import("services").Services} Services
+ */
+
+//-----------------------------------------------------------------------------
+// Internals
+//-----------------------------------------------------------------------------
+
+/**
+ * TODO: remove when services do not have side effects anymore
+ * This forsaken block of code ensures that all are properly cleaned up after each
+ * test because they were populated during the starting process of some services.
+ *
+ * @param {typeof registry} registry
+ */
+const __monitorRegistry = (registry) => {
```

Comment:
keep an equivalent function when services don't have side effects.
Some registries are tested (command Registries), reverting adding/removing entries from them is still a feature we want in tests